### PR TITLE
GET /v1/perspectives?fields=name should NOT include the lens association

### PR DESCRIPTION
**Pull request by iamigo**

### CODE REVIEW

File: api/v1/helpers/verbs/utils.js [hunk 1]
```diff
@@ -371,18 +372,37 @@ function deleteAJsonArrayElement(jsonArray, elementName) {
...
-      const f = fields[i];
-      if (props.fieldScopeMap && props.fieldScopeMap[f]) {
-        scopes.push(props.fieldScopeMap[f]);
+    if (props.fieldAbsenceScopeMap) {
+      const keys = Object.keys(props.fieldAbsenceScopeMap);
+      for (let i = 0; i < keys.length; i++) {
+        const fieldName = keys[i];
+        if (fields.indexOf(fieldName) === NOT_FOUND) {
+          const scopeName = props.fieldAbsenceScopeMap[fieldName];
+          scopes.push(scopeName);
```
Comment (shriramshankar) [inline api/v1/helpers/verbs/utils.js]: Do we need to check if props.fieldAbsenceScopeMap[fieldName] returns null or not?

Comment (iamigo) [inline api/v1/helpers/verbs/utils.js]: I'll add that.

**Comment by pallavi2209** [comment]
@iamigo merge the master changes?

File: api/v1/helpers/verbs/utils.js (hunk 1 above)
Comment (pallavi2209) [inline api/v1/helpers/verbs/utils.js]: Given the fieldName is coming from fieldAbsenceScopeMap only, I dont think the check for props.fieldAbsenceScopeMap[fieldName] was necessary.

**Comment by pallavi2209** [comment]
@iamigo So with these changes GET /v1/perspectives?fields=name do not return lens association part, which is this one:
screen shot 2016-10-13 at 3 15 34 pm
But do you want http://localhost:3000/v1/perspectives/:key also not return lens association part??
If yes, then why not change default scope only? Sorry, I am a bit confused on intentions here.

**Comment by iamigo** [comment]
If the user doesn't specify ?fields=... then I want to return the extra lens fields, but if they are explicitly narrowing the list of fields, then I don't want to add those fields since the user hasn't explicitly asked for them.

**Comment by pallavi2209** [comment]
And what about GET /v1/perspectives/:key. With this change, the response does not include lens fields anymore. Do you intend to make that change?

**Comment by iamigo** [comment]
@pallavi2209 good catch that was not my intention, adding test and fixing now

**Comment by iamigo** [comment]
thank you for your diligence looking at this, @pallavi2209!

**Pull request merged**